src/mono/CMakeLists.txt

```diff
@@ -358,6 +358,11 @@ if(NOT "${MSVC_C_ARCHITECTURE_ID}" STREQUAL "")
   set(CMAKE_SYSTEM_PROCESSOR "${MSVC_C_ARCHITECTURE_ID}")
 endif()
 
+if(HOST_BROWSER)
+  # CMAKE_SYSTEM_PROCESSOR is set to x86 by emscripten
+  set(CMAKE_SYSTEM_PROCESSOR "wasm")
```

Comment:
is this really the case? we're checking `elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "wasm" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "wasm32")` on L389, does that mean that didn't work today?